### PR TITLE
feat: adds twistcli scan to buildspec

### PR DESCRIPTION
**Pull request by gduncklee**
Adds a twistcli integration into our Codebuild process. The scan runs on the built image and only pushes if the image is compliant with Prisma vulnerability and compliance policies.

## Changes

- Adds twistcli to our codebuild process

## How to test this PR

1. Ran Codebuild process with the updated buildspec. Ensured email was delivered if an image violated our policy.

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -1,11 +1,29 @@
 version: 0.2
 
+env:
+  variables:
+    SMTP_CREDS_SECRET: cfpb/team/regtech/smtp-ses-creds
+    SMTP_FROM_ADDRESS: no-reply@cfpb.gov
```
Comment (hkeeler) [inline buildspec.yml:6]: For whatever reason, emails from `no-reply` get filtered for me. :shrug: 

```suggestion
    SMTP_FROM_ADDRESS: noreply@cfpb.gov
```

File: buildspec.yml [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
 phases:
   pre_build:
     commands:
       # Set envvars dependent on CodeBuild project's own envvars
+      - sudo yum -y install mailx coreutils --allowerasing --skip-broken
+      - export JOB_NAME=$CODEBUILD_BUILD_ID
+      - export IMAGE_NAME="cfpb/${TEAM_NAMESPACE}/sbl-frontend"
       - export IMAGE_NAME="cfpb/${TEAM_NAMESPACE}/sbl-frontend"
-      - export IMAGE_TAG="preview"
+      - export IMAGE_TAG="twistlock-cli-codebuild"
```
Comment (hkeeler) [inline buildspec.yml:24]: Are we going to switch this back before merge?

File: buildspec.yml [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+    SMTP_CREDS_SECRET: cfpb/team/regtech/smtp-ses-creds
+    SMTP_FROM_ADDRESS: no-reply@cfpb.gov
+  secrets-manager:
+    TL_CONSOLE_URL: cfpb/team/regtech/twistlock:TL_CONSOLE_URL
+    TL_USER: cfpb/team/regtech/twistlock:TL_USER
+    TL_PASSWORD: cfpb/team/regtech/twistlock:TL_PASSWORD
+    SMTP_USERNAME: "${SMTP_CREDS_SECRET}:username"
+    SMTP_PASSWORD: "${SMTP_CREDS_SECRET}:password"
+    SMTP_SERVER: "${SMTP_CREDS_SECRET}:smtp_server"
+    EMAIL_LIST: "${SMTP_CREDS_SECRET}:email_list"
```
Comment (hkeeler) [inline buildspec.yml:14]: Did you add these to the Secret? I don't think we're going to have these in all environments. We may need to come up with another way of getting these values when we move out of dev.

File: buildspec.yml [hunk 4]
```diff
@@ -18,7 +36,33 @@ phases:
...
+      # Setting pipefail preserves the exit code of the following command
+      - set -o pipefail
+      - >
+        ./twistcli images scan --details -address "${TL_CONSOLE_URL}"  \
+          -u "${TL_USER}" \
+          -p "${TL_PASSWORD}" \
+          "${ECR_ACCOUNT_REGISTRY}/${IMAGE_NAME}:${IMAGE_TAG}" | tee twistcli.log; EXITCODE=$?
+      - >
+        if [ "$EXITCODE" -ne 0 ]; then
+          cat -v twistcli.log | tr -d '^[[' > twistcli-parsed.log
```
Comment (hkeeler) [inline buildspec.yml:51]: Could you add a comment about why we need the `tr` bit?

File: buildspec.yml [hunk 5]
```diff
@@ -18,7 +36,33 @@ phases:
...
+          LOG=$(cat twistcli-parsed.log)
+          echo $LOG | 
+          mailx -s "Twistlock Policy Violation $JOB_NAME" \
+            -S smtp-use-starttls \
+            -S ssl-verify=ignore \
+            -S smtp-auth=login \
+            -S smtp=smtp://"$SMTP_SERVER":587 \
```
Comment (hkeeler) [inline buildspec.yml:58]: Be good if we could get that port into a Secret too.

**Review by hkeeler — COMMENTED** [review]
Looks good. Just a few questions/comment. Let's chat tomorrow.

File: buildspec.yml (hunk 3 above)
Comment (gduncklee) [reply buildspec.yml]: Yes, I didn't want our email addresses exposed. Agreed that we can readdress this when we move out of dev.

File: buildspec.yml (hunk 2 above)
Comment (gduncklee) [reply buildspec.yml]: Good catch, thanks. Updated.

**Review by hkeeler — APPROVED** [review]

**Comment by billhimmelsbach** [comment]
Feel free to merge away! 🎉

**Pull request merged**